src/test/java/teammates/test/cases/ui/pagedata/InstructorFeedbackEditPageDataTest.java

```diff
@@ -0,0 +1,322 @@
+        assertEquals(TimeHelper.formatDate(fs.resultsVisibleFromTime), additionalSettings.getResponseVisibleDateValue());
+        assertEquals(TimeHelper.formatDate(fs.sessionVisibleFromTime), additionalSettings.getSessionVisibleDateValue());
+        
+        assertFalse(additionalSettings.isResponseVisiblePublishManuallyChecked());
+        assertTrue(additionalSettings.isResponseVisibleDateChecked());
+        assertFalse(additionalSettings.isResponseVisibleImmediatelyChecked());
+        assertFalse(additionalSettings.isResponseVisibleNeverChecked());
+        assertFalse(additionalSettings.isResponseVisibleDateDisabled());
+       
+        assertFalse(additionalSettings.isSessionVisibleAtOpenChecked());
+        assertFalse(additionalSettings.isSessionVisibleDateDisabled());
+        assertTrue(additionalSettings.isSessionVisibleDateButtonChecked());
+        assertFalse(additionalSettings.isSessionVisiblePrivateChecked());
+        assertTrue(additionalSettings.isSendClosingEmailChecked());
+        assertTrue(additionalSettings.isSendOpeningEmailChecked());
+        assertTrue(additionalSettings.isSendPublishedEmailChecked());
+        
+        // test question edit forms
+        List<FeedbackQuestionEditForm> questionForms = data.getQnForms();
+        assertEquals(3, questionForms.size());
+        assertEquals(Const.ActionURIs.INSTRUCTOR_FEEDBACK_QUESTION_EDIT, questionForms.get(0).getAction().toString());
+        assertEquals(fs.courseId, questionForms.get(0).getCourseId());
+        assertEquals(fs.feedbackSessionName, questionForms.get(0).getFeedbackSessionName());
+
+        assertEquals(3, questionForms.get(0).getNumOfQuestionsOnPage());
+        assertEquals(dataBundle.feedbackQuestions.get("qn1InSession1InCourse1"), questionForms.get(0).getQuestion());
+        assertEquals(3, questionForms.get(0).getQuestionNumberOptions().size());
+        assertEquals("What is the best selling point of your product?", questionForms.get(0).getQuestionText());
+        assertEquals(false, questionForms.get(0).getFeedbackPathSettings().isNumberOfEntitiesToGiveFeedbackToChecked());
+        assertEquals(1, questionForms.get(0).getFeedbackPathSettings().getNumOfEntitiesToGiveFeedbackToValue());
+        assertEquals(8, questionForms.get(0).getFeedbackPathSettings().getRecipientParticipantOptions().size());
+        assertEquals(4, questionForms.get(0).getFeedbackPathSettings().getGiverParticipantOptions().size());
+        assertEquals("-1", questionForms.get(0).getQuestionNumberSuffix());
+        
+        assertTrue(questionForms.get(0).getVisibilitySettings().getResponseVisibleFor().get(FeedbackParticipantType.INSTRUCTORS.name()));
+        assertFalse(questionForms.get(0).getVisibilitySettings().getResponseVisibleFor().containsKey(FeedbackParticipantType.RECEIVER.name()));
+        assertFalse(questionForms.get(0).getVisibilitySettings().getResponseVisibleFor().containsKey(FeedbackParticipantType.STUDENTS.name()));
```

Comment:
Use variables for `questionForms.get(*).getVisibilitySettings() / getFeedbackPathSettings()`?
They seem to be used often enough to warrant that.